### PR TITLE
layers: Handle OpTypeBool in structs

### PR DESCRIPTION
**Pull request by sjfricke**
address issue in https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/4255#issuecomment-1219214198

### CODE REVIEW

File: layers/shader_module.cpp [hunk 1]
```diff
@@ -1906,6 +1907,9 @@ uint32_t SHADER_MODULE_STATE::GetBaseType(const spirv_inst_iter &iter) const {
         const auto& type = get_def(iter.word(3));
         return GetBaseType(type);
     }
+    // if we assert, we are missing a valid base type and need to handle or else returning zero will produce a hard bug to track
+    // down as a value of zero means nothing useful
```
Comment (ncesario-lunarg) [inline layers/shader_module.cpp:1911]: Had to read this a couple times to understand it. Would this also be accurate to say?
```
If we assert here, we are missing a valid base type that must be handled. Without this assert, a return value of 0 will produce a hard bug to track...
```
This also makes me wonder if we can/should assert in `SHADER_MODULE_STATE::get_def` that `id != 0`, but I didn't see anything explicitly stating result IDs in SPIRV must be > 0.

**Review by ncesario-lunarg — APPROVED** [review]
One comment nit and question about SPIR-V IDs, but otherwise LGTM!

**Review by Tony-LunarG — APPROVED** [review]
LGTM and fixes the CTS tests I saw problems with

File: layers/shader_module.cpp (hunk 1 above)
Comment (sjfricke) [reply layers/shader_module.cpp]: > Would this also be accurate to say?

Yes, will update

> if we can/should assert in SHADER_MODULE_STATE::get_def

I guess I have no idea if anywhere in the code is relying on `get_def` to return `0` to know it is invalid... I think this is a bigger issue to tackle, I do see things like 

```cpp
auto def = get_def(id);
assert(def != end());
```

but also see things like `GetConstantDef` check and pass back the `end()` to it's caller

I think before we assert at that level, someone (happy to do) will need to examine the 113 cases that currently call it and make sure... might be worth just creating a proper issue for this

**Pull request merged**